consistent_df/enforce_schema.py

```diff
@@ -0,0 +1,122 @@
+import re
+from zoneinfo import ZoneInfo
+import pandas as pd
+
+
+def enforce_schema(
+    data: pd.DataFrame,
+    schema: pd.DataFrame,
+    keep_extra_columns: bool = False
+) -> pd.DataFrame:
+    """Ensures that a DataFrame adheres to a specified column schema, enforcing
+    the presence and data types of specified columns.
+
+    This function validates and adjusts a DataFrame to follow a predefined schema.
+    The schema is provided as a separate DataFrame, where each row specifies a column name,
+    its required data type, and whether it's mandatory. It guarantees that required columns
+    are present, adds optional columns if missing, ensures the correct data types, and orders
+    the columns as defined in the schema.
```

Comment:
Description can be clarified a shortened. Please remove redundancy with argument descriptions.